### PR TITLE
chore: bump flask-caching

### PR DESCRIPTION
**Pull request by EugeneTorap**
#21228 & #21555 have been superseded by this PR.

### SUMMARY
Bump the next libs:

- cachelib to 0.9.0
- flask-caching to 2.0.2
- redis to 4.5.1

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by john-bodley** [comment]
@EugeneTorap I realize that the title is somewhat self explanatory but would you mind  filling out the PR description.

Knowing how you bumped these requirements, manually versus using `pip-compile-multi` (including the command if it was the later) would be helpful for reviewing.

Note all Python requirements (and their sub requirements) should be managed via `pip-compile-multi`. Furthermore if these requirements which have been bumped (used primarily for deterministic CI) require a code change then the eligible limits should be updated in `setup.py`.

File: tests/integration_tests/cachekeys/api_tests.py [hunk 1]
```diff
@@ -149,9 +149,9 @@ def test_invalidate_existing_caches(invalidate):
     )
 
     assert rv.status_code == 201
-    assert cache_manager.cache.get("cache_key1") is None
-    assert cache_manager.cache.get("cache_key2") is None
-    assert cache_manager.cache.get("cache_key4") is None
+    assert cache_manager.cache.get("cache_key1") == "value"
+    assert cache_manager.cache.get("cache_key2") == "value"
+    assert cache_manager.cache.get("cache_key4") == "value"
```
Comment (EugeneTorap) [inline tests/integration_tests/cachekeys/api_tests.py:154]: @dpgaspar @etr2460 @bkyryliuk  @Antonio-RiveroMartnez @Yoann-tildev Do you know why this `cache_manager` behavior has changed in new version of `flask-caching`?

Comment (dpgaspar) [reply tests/integration_tests/cachekeys/api_tests.py:154]: no idea, have you debugged it

Comment (EugeneTorap) [reply tests/integration_tests/cachekeys/api_tests.py:154]: Hi @bkyryliuk! I see you created this unit test. Is this change critical for superset?

**Pull request closed without merging**